File: Bully/bully.c
#include <stdio.h>

int maximum(int a,int b){
  if (a > b){
    return a;
  } else {
    return b;
  }
  return a;
}

void main(){
  //Sort the array first
  // int n;
  // printf("Enter the number of nodes");
  // scanf("%d",n);

  int nodes[] = {1,2,3,4,5,6,7};
  // int nodes[n];
  // for(int i = 0; i < n; i++){
  //   printf("enter the %d node",i);
  //   scanf("%d",nodes[i]);
  // }
  

  int start =  2;
  size_t n = sizeof(nodes)/sizeof(nodes[0]);

  int state[] = {1,1,1,1,1,1,0};
  size_t s = sizeof(state)/sizeof(state[0]);
  int max;
  int count = 0;
  for(int i = 0; i<n; i++){
    if(nodes[start] < nodes[i]){
      count++;
    }
  }
  while(count > 0){

    // sending message
    for(int i = 0; i<n; i++){count
        
      if(nodes[start] < nodes[i]){
        printf("Node %d ",nodes[start]);printf("sends message to %d \n",nodes[i]);
      }
    }
    // reply
    for(int i = 0; i < s; i++){
      // printf("%d",state[i]);
      if(state[i] == 1 && nodes[start] < nodes[i]){
        printf("Node %d",nodes[i]);printf(" replies ok \n");
        max = maximum(max, nodes[i]);
      }
    }
    count--;
    start++;
  }

 printf("%d is the new leader ",max);

}

File: Bully/bully.java
import java.util.Scanner;

/**
 * bully
 */
public class bully {

    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        System.out.println("Enter the length of the array:");
        int length = s.nextInt();
        int[] nodes = new int[length];
        int[] state = new int[length];
        System.out.println("Enter the elements of the array:");

        for (int i = 0; i < length; i++) {
            nodes[i] = s.nextInt();
        }
        System.out.println("Enter the state of the nodes:");

        for (int i = 0; i < length; i++) {
            state[i] = s.nextInt();
        }

        // System.out.println(Arrays.toString(nodes));
        // System.out.println(Arrays.toString(state));

        // System.out.println("Which nodes do you want to fail");
        // int fail = s.nextInt();

        System.out.println("Enter the discovering node");
        int start = s.nextInt();

        int max = 0,count = 0;

        for(int i = 0; i < nodes.length; i++){
            if(nodes[start] < nodes[i]){
                count++;
            }
        }

        while(count > 0){
            for(int i = 0; i < nodes.length; i++){
                if(state[start] == 1 && nodes[start] < nodes[i]){
                    System.out.print(nodes[start]+" Sends message to ");
                    System.out.print(nodes[i]);
                    System.out.println("");
                }
            }
            for(int i = 0; i < nodes.length; i++){
                if(state[start] == 1 && state[i] == 1 && nodes[start] < nodes[i]){
                    System.out.print(nodes[i]+" Replies with OK to ");
                    System.out.println(nodes[start]);

                    max= Math.max(max, nodes[i]);
                }
            }
            count--;
            start++;
        }
        System.out.println(max + " Is the new Cordinator");
       s.close();
    }
}
